### PR TITLE
Add `Participant::Resume` service and endpoints

### PR DESCRIPTION
**Pull request by ethax-ross**
[Jira-3088](https://dfedigital.atlassian.net.mcas.ms/jira/software/projects/CPDLP/boards/87?assignee=712020%3A5b5839dc-e4d6-4df5-9dd6-627cc49fe345&selectedIssue=CPDLP-3088)

### Context

We want to add the participant resume endpoint from ECF, bringing across anything relevant to NPQ.

### Changes proposed in this pull request

- Add Participant::Resume service

Add the service to move a participant back to the `active` state. This is lifted from ECF; instead of looking for the matching profile we find the matching application.

- Add /participants/npq/:id/resume endpoint

Adds the endpoint to enable providers to resume the training status of a participant.

### CODE REVIEW

File: app/controllers/api/v1/participants_controller.rb [hunk 1]
```diff
@@ -27,7 +36,18 @@ def participants_query
       end
 
       def participant_params
-        params.permit(:ecf_id, filter: %i[updated_since])
```
Comment (cwrw) [inline app/controllers/api/v1/participants_controller.rb:30]: could we keep the updated since here?

Comment (ethax-ross) [reply app/controllers/api/v1/participants_controller.rb:30]: It doesn't appear to be used? we pull in a concern which exposes the `updated_since`:

https://github.com/DFE-Digital/npq-registration/blob/main/app/controllers/api/v1/participants_controller.rb#L5
https://github.com/DFE-Digital/npq-registration/blob/main/app/controllers/api/concerns/filter_by_updated_since.rb#L8

File: spec/support/shared_examples/api_defer_participant_endpoint_support.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples "an API defer participant endpoint" do
```
Comment (cwrw) [inline spec/support/shared_examples/api_defer_participant_endpoint_support.rb:3]: ✂️ 😅

**Review by cwrw — APPROVED** [review]

File: spec/support/shared_examples/api_defer_participant_endpoint_support.rb (hunk 2 above)
Comment (ethax-ross) [reply spec/support/shared_examples/api_defer_participant_endpoint_support.rb:3]: Whoops, juggling too many PRs 😬

**Pull request merged**